File: remove-duplicates-from-sorted-array-ii/removeDuplicates.go
func removeDuplicates(nums []int) int {
    var result, pointer, count int
    previous :=  math.MinInt32

    for _, num := range nums {
        if num != previous || count < 2 {
            nums[pointer] = num
            pointer++
            if num != previous {
                previous = num
                count = 0
            }
            result++
        }
        count++
    }

    return result
}

File: remove-duplicates-from-sorted-array/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let p = 0,
    result = 0,
    previous;
  nums.forEach((element) => {
    if (element != previous) {
      nums[p++] = element;
      previous = element;
      result++;
    }
  });
  return result;
};
